nbodykit/transform.py

```diff
@@ -486,6 +486,36 @@ def mass_to_radius(mass, redshift):
 
     return da.map_blocks(mass_to_radius, mass, redshift, dtype=mass.dtype)
 
+
+def VectorProjection(vector, direction):
+    """
+    Vector component of a given vector in a given direction.
+
+    .. math::
+
+        \mathbf{v}_d &= (\mathbf{v} \cdot \hat{\mathbf{d}}) \hat{\mathbf{d}} \\
+        \hat{\mathbf{d}} &= \frac{\mathbf{b}}{\|\mathbf{b}\|}
+
+    Parameters
+    ----------
+    vector : array_like, (..., D)
+        array of vectors to be projected
+    direction : array_like, (D,)
+        projection direction. It does not have to be normalized
+
+    Returns
+    -------
+    projection : array_like, (..., D)
+        vector component of the given vector in the given direction
+    """
+    direction = numpy.asarray(direction)
+    direction = direction / numpy.sqrt(direction ** 2)
+    projection = numpy.dot(vector, direction)[..., None]
```

Comment:
To use dask, I suppose you can do 
```
projection = (vector * direction[None, ...]).sum(axis=-1)
projection = projection[..., None] * direction[None, :]
```
We'll also add a unit test for this. Let me find out how to edit a PR

Reply:
I added a unit test. There was actually a bug,